### ISSUE TITLE
Updater: download-progress event not firing

### ISSUE THREAD

**Issue by Christilut** [electron-userland/electron-builder#4919]
* **Version**: 
    "vue-cli-plugin-electron-builder": "1.4.4",
    "electron-builder": "^21.2.0",

* **Electron Version**: 
    "electron": "^6.1.7",

* **Electron Type (current, beta, nightly)**: current

* **Target**: Windows

From the autoUpdater logs:
```
[2020-05-02 10:59:11.283] [info] Checking for update
[2020-05-02T08:59:11.290Z] [verbose]: updater: checking-for-update
[2020-05-02T08:59:11.291Z] [verbose]: status message to renderer, win (true): {"loading":true,"message":"Checking for update..."}
[2020-05-02T08:59:11.291Z] [debug]: application_update event: {"loading":true,"message":"Checking for update..."}
[2020-05-02 10:59:14.436] [info] Found version 1.2.37 (url: Rekordcloud-Companion-App-Setup.exe)
[2020-05-02T08:59:14.441Z] [verbose]: updater: update-available
[2020-05-02 10:59:14.441] [info] Downloading update from Rekordcloud-Companion-App-Setup.exe
[2020-05-02T08:59:14.442Z] [verbose]: status message to renderer, win (true): {"loading":true,"message":"Update found, downloading..."}
[2020-05-02T08:59:14.442Z] [debug]: application_update event: {"loading":true,"message":"Update found, downloading..."}
[2020-05-02 10:59:14.448] [debug] updater cache dir: C:\Users\Alpha\AppData\Local\rekordcloud-companion-app-updater
[2020-05-02 10:59:14.456] [info] No cached update info available
[2020-05-02 10:59:14.462] [info] Download block maps (old: "https://github.com/rekordcloud/companion-app-releases/releases/download/v1.2.36/Rekordcloud-Companion-App-Setup.exe.blockmap", new: https://github.com/rekordcloud/companion-app-releases/releases/download/v1.2.37/Rekordcloud-Companion-App-Setup.exe.blockmap)
[2020-05-02 10:59:15.443] [debug] g37lYn5+Uv1hYjkjPz8yNqpB duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
[2020-05-02 10:59:15.449] [debug] 9sp5PQVmpRErRDAGvBxKRxuc duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
[2020-05-02 10:59:15.454] [debug] wYAigekaq26SS4iYDHzXIMtX duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
... more of the same
[2020-05-02 10:59:15.927] [info] File has 3972 changed blocks
[2020-05-02 10:59:15.934] [debug] [
  {
    "kind": 1,
    "start": 0,
    "end": 15450
  },
  {
    "kind": 0,
    "start": 15450,
    "end": 23693
  },
  ... more of the same
]
[2020-05-02 10:59:15.954] [info] Full: 197,721.73 KB, To download: 88,373.3 KB (45%)
[2020-05-02 10:59:16.710] [info] Differential download: https://github.com/rekordcloud/companion-app-releases/releases/download/v1.2.37/Rekordcloud-Companion-App-Setup.exe
[2020-05-02 10:59:16.716] [debug] download range: bytes=0-15449
[2020-05-02 10:59:17.035] [info] Redirect to https://github-production-release-asset-2e65be.s3.amazonaws.com/211889690/3513c780-8b98-11ea-90e7-2ed003eb3083
[2020-05-02 10:59:17.145] [debug] download range: bytes=23693-80118
[2020-05-02 10:59:17.265] [debug] download range: bytes=126502-139280
[2020-05-02 10:59:17.404] [debug] download range: bytes=8443316-12174819
[2020-05-02 10:59:18.540] [debug] download range: bytes=25503850-26079736
... more of this
```

During all this the `download-progress` event has not fired.

```ts
    autoUpdater.on('checking-for-update', () => {
      log('updater: checking-for-update')
    })

    autoUpdater.on('update-available', async () => {
      log('updater: update-available')
    })

    autoUpdater.on('download-progress', (progressObj: IDownloadProgress) => { // Does not fire
      log('updater: download-progress')
    })
```

It has worked before and I can't really say if it's broken for all users or not but this issue has been in here for quite a while.

**Comment by futurelucas4502** [comment]
I have the same issue its weird as sometimes it will fire and sometimes it won't

**Comment by Christilut** [comment]
Yes, still relevant. Occurs occasionally, can't really reproduce this reliably

**Comment by harshitsilly** [comment]
@Christilut @futurelucas4502  Hi, try this module [electron-differential-updater](https://imjselectron.github.io/electron-differential-updater/#/).. it is built on top of electron-builder and help u with the fix.
It also has differential update for mac also.

**Comment by amiller-gh** [comment]
@harshitsilly should this be contributed back to electron-updater?

**Comment by harshitsilly** [comment]
@amiller-gh - Yeah it can be contributed back to electron-builder. But as we are having more functionalities like differential updater for mac(zip) , dmg(upcoming) and bug fixes regarding differential updater for windows.. we find it useful to create a fork of this module and do that changes there and published it since creating pr and merging can take lot of time and for feature like differential update for mac we are uncertain that pr will be merged back to electron-updater.

**Comment by Oscabrera** [comment]
the solution is use electron-differential-updater

https://www.npmjs.com/package/electron-differential-updater

**Comment by harshitsilly** [comment]
@Oscabrera  - hi plz use this module https://www.npmjs.com/package/@imjs/electron-differential-updater 
the one you are referring to is not supported by us anymore..

**Comment by samuelmaddock** [comment]
This issue seemingly started happening when differential downloads were fixed (#4736). When the differential download failed previously, the full download would occur and 'download-progress' events would be emitted as expected. With differential downloads working, the events are no longer emitted.

**Comment by faribauc** [comment]
I'm stuck on this as well... Linux build doesn't support diff download so it works fine, but Win32 build triggers only `update-available`.
